reframe/core/schedulers/slurm.py

```diff
@@ -692,3 +724,316 @@ def _extract_attribute(self, attr_name, node_descr, sep=None):
 
     def __str__(self):
         return self._name
+
+
+@register_scheduler('slurmfc')
+class SlurmFirecrestJobScheduler(SlurmJobScheduler):
+    def __init__(self, *args, **kwargs):
+        def set_mandatory_var(var):
+            res = os.environ.get(var)
+            if res:
+                return res
+
+            raise JobSchedulerError(f'the env var {var} is mandatory for the '
+                                    f'firecrest scheduler')
+
+        if sys.version_info < (3, 7):
+            raise JobSchedulerError('the firecrest scheduler needs '
+                                    'python>=3.7')
+
+        super().__init__(*args, **kwargs)
+        client_id = set_mandatory_var("FIRECREST_CLIENT_ID")
+        client_secret = set_mandatory_var("FIRECREST_CLIENT_SECRET")
+        token_uri = set_mandatory_var("AUTH_TOKEN_URL")
+        firecrest_url = set_mandatory_var("FIRECREST_URL")
+        self._system_name = set_mandatory_var("FIRECREST_SYSTEM")
+        self._remotedir_prefix = set_mandatory_var('FIRECREST_BASEDIR')
```

Comment:
> How persistent are those variables? Does it suffice to define them once in the config or are they meant to change on every reframe invocation?

Yes, it is enough to define them once in the config. I thought the client ID and secret should come from env vars instead of being hardcoded but I guess we do that in the config anyway.

> I suggest that you make them `sched_options` in the configuration.

The thing that I don't like is that these are more per system than per partition https://reframe-hpc.readthedocs.io/en/latest/config_reference.html#config.systems.partitions.sched_options but it could work still.